test/libs/helper-functions.js

```diff
@@ -0,0 +1,194 @@
+          unregisterPromises.push(
+            registrations[i].unregister()
+          );
+        }
+        return Promise.all(unregisterPromises);
+      });
+  },
+
+  clearAllCaches: function() {
+    return window.caches.keys()
+      .then(cacheNames => {
+        if (cacheNames.length === 0) {
+          return Promise.resolve();
+        }
+
+        var cacheTaskPromises = [];
+        for (var i = 0; i < cacheNames.length; i++) {
+          cacheTaskPromises.push(window.caches.delete(cacheNames[i]));
+        }
+        return Promise.all(cacheTaskPromises);
+      });
+  },
+
+  // Waiting for a service worker to install is handy if you only care
+  // about testing events that have occured in the install event
+  installSW: function(swUrl) {
+    return new Promise((resolve, reject) => {
+      var iframe;
+      getIframe()
+      .then(newIframe => {
+        var options = null;
+        if (newIframe) {
+          options = {scope: iframe.contentWindow.location.pathname};
+          iframe = newIframe;
+        }
+
+        return navigator.serviceWorker.register(swUrl, options);
+      })
+      .then(registration => {
+        if (registration.installing === null) {
+          throw new Error(swUrl + ' already installed.');
+        }
+
+        // We unregister all service workers after each test - this should
+        // always trigger an install state change
+        registration.installing.onstatechange = function() {
+          if (this.state !== 'installed') {
+            return;
+          }
+
+          resolve(iframe);
+        };
+      })
+      .catch(err => {
+        reject(err);
+      });
+    });
+  },
+
+  // To test fetch event behaviour in a service worker you will need to wait
+  // for the service worker to activate
+  activateSW: function(swUrl) {
+    return new Promise((resolve, reject) => {
+      var iframe;
+      getIframe()
+      .then(newIframe => {
+        var options = null;
+        if (newIframe) {
+          options = {scope: newIframe.contentWindow.location.pathname};
+          iframe = newIframe;
+        }
+        return navigator.serviceWorker.register(swUrl, options);
+      })
+      .then(registration => {
+        if (registration.installing === null) {
+          throw new Error(swUrl + ' already installed.');
+        }
+
+        // We unregister all service workers after each test - so this should
+        // always have an activate event if the service worker calls
+        // self.clients.claim()
+        registration.installing.onstatechange = function() {
+          if (this.state !== 'activated') {
+            return;
+          }
+
+          resolve(iframe);
+        };
+      })
+      .catch(err => {
+        reject(err);
+      });
+    });
+  },
+
+  // This is a helper method that checks the cache exists before
+  // getting all the cached responses.
+  // This is limited to text at the moment.
+  getAllCachedAssets: function(cacheName) {
```

Comment:
`getAllCachedAssets()` resolves to an `Object` with keys that consist of URLs, and values that consist of `Response`s. As far as I can tell, though, you never make use of the `Response` values anywhere, and just use `Object.keys()` to extract out the URLs from the resolved value.

Can you just use an array of URLs as the resolution value? You can get those pretty easily via `cache.keys()` and save yourself a lot of trouble.